File: app/src/main/java/com/p1/uber/provides/TokenProvider.java
package com.p1.uber.provides;

import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.InstanceIdResult;
import com.p1.uber.modelos.Token;

public class TokenProvider {

    DatabaseReference mDatabase;

       public TokenProvider() {
        mDatabase = FirebaseDatabase.getInstance().getReference().child("Tokens");

    }

    public void create (final String idUser){
           if (idUser == null) return;
        FirebaseInstanceId.getInstance().getInstanceId().addOnSuccessListener(new OnSuccessListener<InstanceIdResult>() {
            @Override
            public void onSuccess(InstanceIdResult instanceIdResult) {
                Token token = new Token(instanceIdResult.getToken());
                mDatabase.child(idUser).setValue(token);
            }
        });
    }

    public  DatabaseReference getTokens(String IdUser){
           return  mDatabase.child(IdUser);
    }
}

File: app/src/main/java/com/p1/uberfares/provides/TokenProvider.kt
package com.p1.uberfares.provides

import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.iid.FirebaseInstanceId
import com.p1.uberfares.modelos.Token

class TokenProvider {
    var mDatabase: DatabaseReference

    init {
        mDatabase = FirebaseDatabase.getInstance().reference.child("Tokens")
    }

    fun create(idUser: String?) {
        if (idUser == null) return
        FirebaseInstanceId.getInstance().instanceId.addOnSuccessListener { instanceIdResult ->
            val token = Token(instanceIdResult.token)
            mDatabase.child(idUser).setValue(token)
        }
    }

    fun getTokens(IdUser: String?): DatabaseReference {
        return mDatabase.child(IdUser!!)
    }
}